### PR TITLE
#14 add Activities and ActivityType models

### LINKED ISSUE #14: Build activity model
### Describe the Feature / Enhancement

We want the user to be able to track their activity for the day, and as such need the Activities model to be built out. We will also need to build the **ActivityTypes** model as Activity model shares a ForeignKey

### Additional Details and Resources

The models should look something like this:

ActivityTypes
id: BigInt (PK)
name: text

Activities
id: BigInt (PK)
type: BigInt (FK: ActivityTypes.id)
user_description: text
duration: integer
difficulty: integer

### Correct Project Selected

- [X] This issue is linked to the appropriate project.

### Correct Label Selected

- [X] This issue has appropriate labels.

### Correct Milestone Selected

- [X] This issue has appropriate milestone.

### PR DESCRIPTION
**Pull request by AidanMargo**
## Description of Proposed Changes

_What does this pull request propose to change?_

Created the models for Activities and ActivityType, and registered them to the admin site to resolve #14  

### Type of Change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Testing

_Please describe the tests that you ran to verify your changes, providing instructions so we can reproduce. Also, please list any relevant details for your test configuration._

Checked models appeared in django admin site.
1. Created an Activity and an ActivityType instance in admin 
2. assigned the Activity to the new Activity Type.
3. Delete the ActivityType instance
4. Activity should still exist, but no longer be assigned to an ActivityType.

## Checklists

_Please ensure that the following items all have checked boxes before setting your pull request as "ready to review". If an item is not relevant to the proposed changes, you may check the corresponding box._

### GitHub

- [x] Related issue appears at beginning of pull request title with pound sign, and title describes the changes being proposed.
- [x] Related issue is linked in pull request description using [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] The appropriate label has been chosen for this pull request.
- [x] The correct project has been selected for this pull request.
- [x] The milestone corresponding with the issue has been selected for this pull request.
- [x] All commits in this branch, including merge commits, begin with the issue number and a pound sign.

### Code Standards

- [x] Code follows the style guidelines of this project.
- [x] Code comments have been added, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] These changes generate no new warnings.

### Documentation and Testing

- [x] New and existing unit tests pass locally with these changes.
- [x] Corresponding changes to the documentation have been made.
- [x] Tests have been added, where appropriate; or, no tests are relevant for this pull request.
- [x] This pull request contains UI changes, and screenshots of those UI images appear below; or, this pull request contains no UI changes.

### CODE REVIEW

File: base_login/__init__.py
Comment (seidior) [inline base_login/__init__.py:1]: Before we do activities and activity types, we should probably add some of the user model stuff first, like goals and goal types.

As for what to call the relevant Django apps, what do you think about:
- "auth" for authentication-related stuff, including account creation, account login, session destruction, and account destruction
- "goals" for goal-setting
- "activities" for activity tracking
- "reminders" for reminder tracking

File: base_login/models.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f'{self.name}'
+
+class Activities(models.Model):
+    name = models.CharField(max_length=255)
+    type = models.ForeignKey('ActivityTypes', on_delete=models.CASCADE)
```
Comment (seidior) [inline base_login/models.py]: I'm not sure we'd want an `ON DELETE CASCADE` here; perhaps just setting it null so we don't delete user data associated with the activity type being deleted.

**Review by seidior — COMMENTED** [review]
Hi @AidanMargo! I know this PR is still in draft so a review is not appropriate yet, but I went ahead and took a look at it for you. I have some notes that you'll see below. I'm most interested in what you think about naming and encapsulation for Django apps; I still want to have that conversation with you whenever you're free.

I'm going to leave this review as comment status rather than request changes since the PR still needs to be updated and brought back to ready for review.

File: base_login/__init__.py
Comment (AidanMargo) [reply base_login/__init__.py:1]: The naming conventions sound good to me. I went ahead and changed the name of this app to "activities."

File: base_login/models.py (hunk 1 above)
Comment (AidanMargo) [reply base_login/models.py:16]: Changed to `on_delete=None`

Comment (AidanMargo) [reply base_login/models.py]: Good point. I went ahead and changed that to `on_delete=models.SET_NULL` as to prevent that from happening.

**Review by seidior — APPROVED** [review]
You're now clear to merge this. Make sure you're checking linting with `prospector` before marking your PR as ready to review; there were a few errors that needed fixing. I didn't want to hold you up on review cycle though so I went in and fixed them for you :)

**Pull request merged**